[DAL] Spec of common types (#222)
oneapi-src/oneAPI-spec: source/elements/oneDAL/source/common_interface/memory_management.rst

=================
Memory management
=================

This section describes common memory management types in |dal_short_name|.

---------------------
Programming interface
---------------------

All types and functions in this section shall be declared in the
``oneapi::dal`` namespace and be available via inclusion of the
``oneapi/dal/memory.hpp`` header file.

Empty delete
------------

> The function object that does nothing and can be used as empty deleter for
> |dal_short_name| objects and their data.

Reviewer: is it important to mention that this object does nothing?

```suggestion
The function object that can be used as an empty deleter for
|dal_short_name| objects and their data.
```

.. onedal_class:: oneapi::dal::empty_delete

Default delete
--------------

> The function objects used to free the array of elements of type ``Data``.

Reviewer: ```suggestion
The function object that is used to free an array of elements of type ``Data``.
```

.. onedal_class:: oneapi::dal::default_delete

> The helper function that creates :expr:`default_delete` object using SYCL* queue.

Reviewer: ```suggestion
The helper function that creates a :expr:`default_delete` object using SYCL* queue.
```
